File: Creational-Patterns/C++/Builder/src/builder/workstation_system_builder.cpp
#include "workstation_system_builder.hpp"


hardware_infos::Memory *builder_pattern::WorkstationSystemBuilder::get_memory()
{
    hardware_infos::Memory *mem = new hardware_infos::Memory(MEMORY_TYPE, MEMORY_SIZE, MEMORY_VOLTAGE);
    return mem;
}

hardware_infos::Processor *builder_pattern::WorkstationSystemBuilder::get_processor()
{
    hardware_infos::Processor *proc = new hardware_infos::Processor(PROCESSOR_TYPE,PROCESSOR_CACHE,PROCESSOR_CORES,PROCESSOR_CLOCK);
    return proc;
}

hardware_infos::HardDisk *builder_pattern::WorkstationSystemBuilder::get_hd()
{
    hardware_infos::HardDisk *hd = new hardware_infos::HardDisk(HARDDISK_TYPE,HARDDISK_STORAGE);
    return hd;
}

File: Creational-Patterns/Python/Builder/builder_pattern/workstation_system_builder.py
#coding: utf-8

from builder_pattern import builder
from hardware import processor,memory,harddisk

class WorkstationSystemBuilder(builder.AbstractBuilder):

    def get_memory(self):
        return memory.Memory("DDR4",16,1.5)

    def get_processor(self):
        return processor.Processor("CISC", 4,8,3.2)

    def get_harddisk(self):
        return harddisk.HardDisk("SSD",256)
